docs/raw/serving-community-history.md

```diff
@@ -84,200 +89,282 @@ Furthermore, it assumes that:
 
 ### Receiving community history archives
 
 Community member nodes go through the following (high level) process to fetch and restore community message histories:
 
-1. User joins community and becomes community member
-2. By joining a community, member nodes automatically subscribe to special magnet link channel provided by the community
-3. Member node requests message history (last 30 days) of community channels from store nodes
-4. Member node receives magnet link message from store nodes
-5. Member node extracts magnet link from message and passes it to torrent client
-6. Torrent client downloads latest message archive index via magnet link
-7. Member node fetches missing archives via torrent
-8. Member node unpacks and decompresses message archive data to then hydrate its local database
+1. User joins community and becomes community member (see [org channels spec](https://github.com/status-im/specs/pull/151))
+2. By joining a community, member nodes automatically subscribe to special channel for message archive metadata exchange provided by the community
+3. Member node requests live message history (last 30 days) of special community channel from store nodes (in addition to last 7 days of message for all other community channels)
+4. Member node receives waku message ([14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/)) that contains the metadata magnet link from the special channel
+5. Member node extracts the magnet link from the waku message and passes it to torrent client
+6. Member node downloads [message archive index](#message-history-archive-index) file and determines which message archives are not downloaded yet (all or some)
+7. Member node fetches missing message archive data via torrent
+8. Member node unpacks and decompresses message archive data to then hydrate its local database, deleting any messages for that community that the database previously stored in the same time range as covered by the message history archive
 
 ## Storing live messages
 
 Community owner nodes MUST store live messages as [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/). This is required to provide confidentiality, authenticity, and integrity of message data distributed via the BitTorrent layer, and later validated by Status nodes when they unpack message history archives.
 
-Community owner nodes SHOULD remove those messages from their local databases after they have been turned into archives and distributed to the BitTorrent network.
+Community owner nodes SHOULD remove those messages from their local databases once they are older than 30 days and after they have been turned into message archives and distributed to the BitTorrent network.
```

Comment:
Are you sure this is correct?  If a community owner node removes all messages older than 30 days from it's local database, how will the community owner be able to access messages older than 30 days using the Status desktop UI?

Reply:
Ah, we had this same discussion in the other PR :D 

See: https://github.com/status-im/specs/pull/162#discussion_r767980234

This is about removing the Waku messages, not the application messages.

Reply:
ahh yes, thanks for reminding me! Ignore this comment ;-)